scopes/ui-foundation/uis/side-bar/component-tree/component-tree.tsx

```diff
@@ -17,10 +17,11 @@ export function ComponentTree({ components, isCollapsed, TreeNode = DefaultTreeN
   const { pathname } = useLocation();
 
   const activeComponent = useMemo(() => {
+    const componentIdUrlRegex = '[\\w\\/-]*[\\w-]';
     const path = pathname?.startsWith('/') ? pathname.substring(1) : pathname;
+    const matcher = path.match(componentIdUrlRegex); // returns just tha part that matches the componentId section without /~compositions etc.
```

Comment:
I'd move the [0] to this line

Reply:
also make a a constant regex out of this, like `componentUrlRegex = new RegExp(componentIdUrlRegex)`

Reply:
```
const match = path.match(componentUrlRegex)?.[0]
```